datamade/code-challenge
# Add linting setup for JavaScript and Python

## Overview

This PR adds JavaScript and Python linting to the code challenge, in order to make way for more substantive discussion on during code review.

### Demo

```bash
call-me-hank:code-challenge hannah$ docker-compose -f docker-compose.yml -f tests/docker-compose.yml run --rm app
Creating code-challenge_app_run ... done
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, sessions
Running migrations:
  No migrations to apply.
parserator_web/views.py:1:1: F401 'usaddress' imported but unused
parserator_web/views.py:6:1: F401 'rest_framework.exceptions.ParseError' imported but unused
parserator_web/views.py:24:16: F821 undefined name 'address_components'
parserator_web/views.py:24:36: F821 undefined name 'address_type'
tests/test_views.py:7:5: F841 local variable 'address_string' is assigned to but never used
tests/test_views.py:14:5: F841 local variable 'address_string' is assigned to but never used
JavaScript linting passed! 👍
========================================= test session starts ==========================================
platform linux -- Python 3.8.7, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- /usr/local/bin/python
cachedir: .pytest_cache
django: settings: parserator_web.settings (from ini)
rootdir: /app, configfile: setup.cfg
plugins: django-3.9.0, mock-3.2.0
collected 2 items

tests/test_views.py::test_api_parse_succeds FAILED

=============================================== FAILURES ===============================================
________________________________________ test_api_parse_succeds ________________________________________

client = <django.test.client.Client object at 0x7f2c95b7e0a0>

    def test_api_parse_succeds(client):
        # TODO: Finish this test. Send a request to the API and confirm that the
        # data comes back in the appropriate format.
        address_string = '123 main st chicago il'
>       pytest.fail()
E       Failed

tests/test_views.py:8: Failed
======================================= short test summary info ========================================
FAILED tests/test_views.py::test_api_parse_succeeds - Failed
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================== 1 failed in 0.47s ===========================================
```

## Testing Instructions

* Follow the README instructions to build and run the app. Confirm they work as advertised (with the understanding that the business logic of the code is not implemented).
* Follow the README instructions to run the tests and confirm you see the default output from the Demo section.
* Add some invalid JavaScript to the `index.js` file and run the tests again. Confirm you see a linting error.
